### PR TITLE
Support `skewness(x)` in Aggregation function

### LINKED ISSUE #12249: Support` skewness(x) ` in Aggregation function
### Is your feature request related to a problem or challenge?

Returns the skewness value calculated from values of a group.

### Describe the solution you'd like

```
D SELECT skewness(col) FROM VALUES (-10), (-20), (100), (1000), (1000) AS tab(col);
┌────────────────────┐
│   skewness(col)    │
│       double       │
├────────────────────┤
│ 0.5745116147533554 │
└────────────────────┘
```

### Describe alternatives you've considered

_No response_

### Additional context

https://docs.databricks.com/en/sql/language-manual/functions/skewness.html
https://duckdb.org/docs/sql/functions/aggregates#skewnessx

### PR DESCRIPTION
**Pull request by dharanad**
## Which issue does this PR close?

Part of #12254
Closes #12249

Implemented by referring to https://github.com/duckdb/duckdb/blob/main/src/core_functions/aggregate/distributive/skew.cpp

## What changes are included in this PR?
Added `skewness` UDAF.
Ref: https://duckdb.org/docs/sql/functions/aggregates#skewnessx

### CODE REVIEW

**Comment by dharanad** [comment]
Planning to add more logic tests.

**Comment by 2010YOUY01** [comment]
> Planning to add more logic tests.

I think we can port tests from duckdb https://github.com/duckdb/duckdb/blob/main/test/sql/aggregate/aggregates/test_skewness.test

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 1]
```diff
@@ -527,6 +528,19 @@ regr_sxy(expression_y, expression_x)
...
 
+### `skewness`
+
+Computes the skewness value.
+
+```
+skewness(expression)
+```
+
+#### Arguments
```
Comment (comphead) [inline docs/source/user-guide/sql/aggregate_functions.md:539]: I just realized we dont have any examples in `aggregate_functions.ms`

**Comment by comphead** [comment]
Basically I'm feeling we have too much of user .md files, duplicating the functions description and poorly connected to each other which confuses a lot. 

@alamb WDYT about having a single file for all SQL functions/scalars/expressions like in https://spark.apache.org/docs/latest/api/sql/index.html with description, examples

Btw Spark holds an internal meta repository that allows such doc to be regenerated from the code, which can be next step

**Comment by alamb** [comment]
> Btw Spark holds an internal meta repository that allows such doc to be regenerated from the code, which can be next step

I agree

What I actually think would be ideal is exactly as @comphead describes and put all the documentation about the functions in the code, and then have a post processing step that auto generates the documentation

This is how the configuration setting documentation is created. Here is the script that does it:
https://github.com/apache/datafusion/blob/main/dev/update_config_docs.sh

BTW I think this is the direction @findepi  is heading with https://github.com/apache/datafusion/pull/12266. I also think GlareDB did something similar, but I can't find the link now.

**Comment by alamb** [comment]
I also wonder if we should consider where to draw the line on features to include (i.e. should we include `skewness` by default in the core?) -- especially as all aggregate functions use the same API, this is the kind of feature we could very easily keep outside -- either in its own crate or even own repo -- and then have other code integrate it in -- e.g. #11979

**Comment by dharanad** [comment]
Closing this PR. Will move this to https://github.com/datafusion-contrib/datafusion-functions-extra as suggested

**Pull request closed without merging**